clang/lib/Sema/SemaTemplateInstantiate.cpp

```diff
@@ -1457,6 +1457,11 @@ static ExprResult TransformUniqueStableName(TemplateInstantiator &TI,
         TI.getSema(), Sema::ExpressionEvaluationContext::Unevaluated);
     ExprResult SubExpr = TI.getDerived().TransformExpr(E->getExpr());
 
+    if (SubExpr.isInvalid())
+      return ExprError();
+
+    SubExpr = TI.getSema().CheckPlaceholderExpr(SubExpr.get());
+
     if (SubExpr.isInvalid())
       return ExprError();
```

Comment:
Looks okay to me.   One question.   Is the "isInvalid()" check necessary after getting the new subexpression?

Reply:
Yes that CheckPlaceholderExpr can fail in any case where the sub expression fails to instantiate (type issues, lookup failures, etc), which will be diagnosed like normal.

Reply:
Got it, thanks.